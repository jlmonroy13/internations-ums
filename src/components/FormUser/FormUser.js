import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FormField from '../FormField';

class FormUser extends Component {
  constructor() {
    super();
    this.state = {
      file: [],
      firstName: '',
      lastName: '',
      email: '',
      group: '',
      facebook: '',
      twitter: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChangeFileInput = this.handleChangeFileInput.bind(this);
  }
  handleSubmit(event) {
    event.preventDefault();
    const { file, group, ...rest } = this.state;
    this.props.createUser(rest, file, group);
  }
  handleChange(event) {
    const { name, value } = event.target;
    this.setState(() => ({
      [name]: value,
    }));
  }
  handleChangeFileInput(event) {
    event.preventDefault();
    const { files } = event.target;
    this.setState(() => ({
      file: files[0],
    }));
  }

  render() {
    const { groups } = this.props;
    function onRenderGroupsOptions(acc, group) {
      if (group && group.id) {
        acc.push(
          <option value={group.id} key={group.id}>
            {group.name}
          </option>,
        );
      }
      return acc;
    }
    return (
      <form className="form" onSubmit={this.handleSubmit}>
        <div className="form__field">
          <label className="form__label-file" htmlFor="photo">
            Load photo
          </label>
          <input
            className="form__input"
            type="file"
            id="photo"
            onChange={this.handleChangeFileInput}
            required={true}
          />
        </div>
        <FormField
          type="text"
          id="firstName"
          name="firstName"
          placeholder="First Name"
          onChange={this.handleChange}
          required={true}
        />
        <FormField
          type="text"
          id="lastName"
          name="lastName"
          placeholder="Last Name"
          onChange={this.handleChange}
          required={true}
        />
        <FormField
          type="email"
          id="email"
          name="email"
          placeholder="email@test.com"
          onChange={this.handleChange}
          required={true}
        />
        <div className="form__field">
          <label className="form__label" htmlFor="group" />
          <div className="form__select-container">
            <select
              className="form__select"
              id="group"
              name="group"
              onChange={this.handleChange}
              required={true}
            >
              <option value="">Select a group</option>
              {groups.reduce(onRenderGroupsOptions, [])}
            </select>
          </div>
        </div>
        <FormField
          type="text"
          id="facebook"
          name="facebook"
          placeholder="Facebook"
          onChange={this.handleChange}
          required={true}
        />
        <FormField
          type="text"
          id="twitter"
          name="twitter"
          placeholder="Twitter"
          onChange={this.handleChange}
          required={true}
        />
        <button className="button button--primary push--top" type="submit">
          Add
        </button>
      </form>
    );
  }
}

FormUser.propTypes = {
  createUser: PropTypes.func.isRequired,
  groups: PropTypes.array,
};

export default FormUser;
