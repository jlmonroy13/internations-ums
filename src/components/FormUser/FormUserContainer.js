import { connect } from 'react-redux';
import { firebaseConnect, dataToJS, pathToJS, isLoaded, isEmpty } from 'react-redux-firebase';
import { createUser } from '../../actions/user';
import FormUser from './FormUser';

const fbWrappedComponent = firebaseConnect(['/groups'])(FormUser);

function mapStateToProps(state) {
  const { firebase } = state;
  const entitiesGroups = pathToJS(firebase, 'profile/groups');
  let items = dataToJS(firebase, 'groups');
  if (isLoaded(items) && !isEmpty(items) && entitiesGroups) {
    items = Object.keys(entitiesGroups).map(groupId => items[groupId]);
  } else {
    items = [];
  }
  return {
    groups: items,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    createUser: (data, file, groupId) => dispatch(createUser(data, file, groupId)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(fbWrappedComponent);
