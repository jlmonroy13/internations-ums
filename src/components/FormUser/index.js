import FormUserContainer from './FormUserContainer';

export default FormUserContainer;
