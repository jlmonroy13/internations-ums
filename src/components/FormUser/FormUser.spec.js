import React from 'react';
import renderer from 'react-test-renderer';
import FormUser from './FormUser';

describe('<FormUser />', () => {
  it('should render the form user component', () => {
    const createUser = jest.fn();
    const wrapper = renderer.create(<FormUser createUser={createUser} groups={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});