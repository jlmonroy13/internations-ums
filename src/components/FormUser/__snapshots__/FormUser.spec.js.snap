// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormUser /> should render the form user component 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="form__field"
  >
    <label
      className="form__label-file"
      htmlFor="photo"
    >
      Load photo
    </label>
    <input
      className="form__input"
      id="photo"
      onChange={[Function]}
      required={true}
      type="file"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="firstName"
    />
    <input
      className="form__input"
      id="firstName"
      name="firstName"
      onChange={[Function]}
      placeholder="First Name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="lastName"
    />
    <input
      className="form__input"
      id="lastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Last Name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="email"
    />
    <input
      className="form__input"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="email@test.com"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="form__label"
      htmlFor="group"
    />
    <div
      className="form__select-container"
    >
      <select
        className="form__select"
        id="group"
        name="group"
        onChange={[Function]}
        required={true}
      >
        <option
          value=""
        >
          Select a group
        </option>
      </select>
    </div>
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="facebook"
    />
    <input
      className="form__input"
      id="facebook"
      name="facebook"
      onChange={[Function]}
      placeholder="Facebook"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form__field"
  >
    <label
      className="sr-only"
      htmlFor="twitter"
    />
    <input
      className="form__input"
      id="twitter"
      name="twitter"
      onChange={[Function]}
      placeholder="Twitter"
      required={true}
      type="text"
    />
  </div>
  <button
    className="button button--primary push--top"
    type="submit"
  >
    Add
  </button>
</form>
`;
